### ISSUE TITLE
Box not detected from alloc crate

### ISSUE THREAD

**Issue by lights0123** [intellij-rust/intellij-rust#3845]
## Environment

* **Intellij-Rust plugin version:** v0.2.97.2123-191
* **Rust toolchain version:** rustc 1.36.0-nightly (372be4f36 2019-05-14) *(but it doesn't really matter)*
* **IDE name and version:** IntelliJ IDEA 2019.1.1 (Ultimate Edition) Build #IU-191.6707.61, built on April 16, 2019
* **Operating system:** Ubuntu & Antergos Linux

## Problem description
When using `extern crate alloc;` (as required in `no_std`), the IDE is completely unaware of anything within [`alloc::boxed`](https://doc.rust-lang.org/nightly/alloc/boxed/index.html). This includes `Box` and `FnBox`.

This seems to be caused by the following lines within `liballoc/lib.rs`:
```rust
// Need to conditionally define the mod from `boxed.rs` to avoid
// duplicating the lang-items when building in test cfg; but also need
// to allow code to have `use boxed::Box;` declarations.
#[cfg(not(test))]
pub mod boxed;
#[cfg(test)]
mod boxed {
    pub use std::boxed::Box;
}
```

## Steps to reproduce
Add the following to a module file:
```rust
extern crate alloc;
use alloc::boxed::Box;
/// This function has no autocomplete or highlighting for `Box`
fn doesnt_work(_param: Box<u8>) {}
/// But this does
fn works(_param: std::boxed::Box<u8>) {}
```

**Comment by vlad20012** [comment]
Unfortunately we don't support `cfg` attributes for now
